cpp/memilio/compartments/feedback_simulation.h

```diff
@@ -0,0 +1,370 @@
+ */
+struct GammaCutOff {
+    using Type = size_t;
+    static Type get_default(AgeGroup)
+    {
+        return 45;
+    }
+    static std::string name()
+    {
+        return "GammaCutOff";
+    }
+};
+
+/**
+ * @brief Maximum allowed contact reduction factors per location.
+ */
+template <typename FP = double>
+struct ContactReductionMax {
+    using Type = std::vector<FP>;
+    static Type get_default(AgeGroup)
+    {
+        return Type(1, FP(1.0));
+    }
+    static std::string name()
+    {
+        return "ContactReductionMax";
+    }
+};
+
+/**
+ * @brief Minimum allowed contact reduction factors per location.
+ */
+template <typename FP = double>
+struct ContactReductionMin {
+    using Type = std::vector<FP>;
+    static Type get_default(AgeGroup)
+    {
+        return Type(1, FP(0.0));
+    }
+    static std::string name()
+    {
+        return "ContactReductionMin";
+    }
+};
+
+/**
+ * @brief Soft-plus curvature parameter for contact adjustment.
+ */
+template <typename FP = double>
+struct SoftPlusCurvatureParameter {
+    using Type = FP;
+    static Type get_default(AgeGroup)
+    {
+        return FP(0.1);
+    }
+    static std::string name()
+    {
+        return "SoftPlusCurvatureParameter";
+    }
+};
+
+/**
+ * @brief Nominal ICU capacity.
+ */
+template <typename FP = double>
+struct NominalICUCapacity {
+    using Type = FP;
+    static Type get_default(AgeGroup)
+    {
+        return FP(9.0);
+    }
+    static std::string name()
+    {
+        return "NominalICUCapacity";
+    }
+};
+
+template <typename FP = double>
+using ParametersFeedback = ParameterSet<ICUOccupancyLocal<FP>, GammaShapeParameter<FP>, GammaScaleParameter<FP>,
+                                        GammaCutOff, ContactReductionMax<FP>, ContactReductionMin<FP>,
+                                        SoftPlusCurvatureParameter<FP>, NominalICUCapacity<FP>>;
+
+/**
+ * @brief A generic feedback simulation extending existing simulations with a feedback mechanism.
+ *
+ * This class wraps any simulation (e.g. Simulation or FlowSimulation) and applies additional
+ * feedback logic in a model-independent way. Model-specific details—such as the ICU compartments—
+ * are provided via arguments.
+ *
+ * @tparam FP The floating point type.
+ * @tparam Sim The simulation type.
+ * @tparam ContactPatterns The model-specific contact patterns type.
+ */
+template <typename FP, typename Sim, typename ContactPatterns>
+class FeedbackSimulation
+{
+public:
+    /**
+     * @brief Constructs the FeedbackSimulation by using a existing simulation instance and ICU compartment indices.
+     *
+     * @param sim The simulation instance to be extended with feedback mechanism.
+     * @param icu_indices A vector of indices indicating ICU compartments for specific model.
+     */
+    explicit FeedbackSimulation(Sim sim, const std::vector<size_t>& icu_indices)
+        : m_simulation(std::move(sim))
```

Comment:
Using both move and pass-by-value is a bit odd, and looking at the example, it seems like you want FbSim to take ownership of sim. In that case, the constructor should take a rvalue ref, i.e. `Sim&& sim`. This also should make the std::move in line 186 here superfluous.